### ISSUE TITLE
Crash in HI miniAOD from PFJetMETcorrInputProducerT.h

### ISSUE THREAD

**Issue by mandrenguyen** [cms-sw/cmssw#33564]
As discussed in 
https://its.cern.ch/jira/browse/PDMVRERECO-43
there is a rare crash in HI miniAOD production.

It occurs in the loop inside the if statement "if(skipMuons_)" on line 181 of
https://cmssdt.cern.ch/lxr/source/JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h?%21v=CMSSW_11_2_1

Looking at one problematic event, modifying the parameter skipMuons to be False (the default value is True), allows the event to complete successfully.

The module which is failing is corrPfMetType1, which takes as input ak4PFJetsCHS, which are clustered from a dummy collection called pfEmptyCollection.

This is intended to be an empty collection of PF Candidates, such that all downstream collections are present but empty.  In the HI miniAOD this strategy is used to deactivate unneeded features, while avoiding crashes in things like DQM modules that might consume the produced collections. 
It's not yet clear why a crash occurs for code that is expected to be inactive.

**Comment by slava77** [comment]
assign reconstruction

**Comment by slava77** [comment]
@mandrenguyen 
please prepare a prescription to reproduce the problem.
Thank you.

**Comment by mandrenguyen** [comment]
I ran workflow 140.5611, using as input a problematic file with the following driver:
`
cmsDriver.py step2 --procModifiers run2_miniAOD_pp_on_AA_103X --conditions auto:run2_data_promptlike_hi -s PAT,DQM:@miniAODDQM --scenario pp --datatier MINIAOD,DQMIO -n 100 --data --era Run2_2018_pp_on_AA --processName PAT --eventcontent MINIAOD,DQM --filein /store/hidata/HIRun2018A/HIMinimumBias13/AOD/04Apr2019-v1/50000/2F9A896F-F3B1-874C-A45C-327FD404B4F7.root --no_exec`

I was too lazy to look up how to apply a lumi mask to the driver so I modified the PoolSource by hand by adding:
                            ```
eventsToProcess = cms.untracked.VEventRange("327004:60462025"),
                            lumisToProcess = cms.untracked.VLuminosityBlockRange("327004:152"),
```

I also check that by adding the following customization, cmsRun completes successfully:  `process.corrPfMetType1.skipMuons = False`

**Comment by slava77** [comment]
@camclean @alefisico @lathomas @kirschen 
please take a look or suggest a JME expert to follow this

**Comment by lathomas** [comment]
Hello. 
I'm digging. In the mentioned event, the jet size here is 1 (while it is zero for usual events).
https://github.com/cms-sw/cmssw/blob/master/JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h#L167 
The jet is apparently made of a single muon of pt = 10 TeV and nothing else...
This can be fixed by checking whether cand->isAvailable() here: 
https://github.com/cms-sw/cmssw/blob/master/JetMETCorrections/Type1MET/interface/PFJetMETcorrInputProducerT.h#L184 

Then I get issues with other modules for a similar reason so a more general fix is needed.

**Comment by lathomas** [comment]
Hello @mandrenguyen 
I think I understand the problem now and it is in fact solved for the future. 

At the moment ak4PFJetsCHS and ak4PFJets are currently not recomputed for MINIAOD production. They are read directly from AOD (on purpose I believe).
Now, the AOD jet collections used to be buggy before the following change happened:
https://github.com/cms-sw/cmssw/pull/30898/files#diff-6b6cc8be88a999f666cc3a98b5067dc853b39d442b282193cd7e14338fb16c17L14-R14
Indeed, before that change, >10 TeV PF candidates (usually badly reconstructed muons) could still make a jet. Because the PF collection is empty in MINIAOD, this causes a crash when one tries to access the jet constituents (the reference cannot be found). 

This will not happen in the future. 

For MINIAOD productions based on existing affected AOD, the problem remains. Here the solution can simply be to recompute the ak4PFJetsCHS and ak4PFJets collections by adding. (I checked that simply adding these two modules to process.schedule allows the problematic events to run fine)

**Comment by mandrenguyen** [comment]
Thank you @lathomas 
A PR to master implementing this fix may be found here:
https://github.com/cms-sw/cmssw/pull/33928

**Comment by slava77** [comment]
+1

fixed in #33928
